tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,724 @@
+            resp = s3_misc.copy_object(self.akey, self.skey, self.src_bkt, "obj1",
+                                       self.src_bkt, dest_obj)
+            assert resp, f"Copy object Failed for {self.src_bkt}/{dest_obj}"
+        self.log.info("Step 7: Perform Copy object operation again (N)th time within same bucket"
+                      " using same source bucket and object and just changing the key name of the"
+                      " destination object .")
+        try:
+            dest_obj = "obj" + str(test_cfg["max_objects"]+1)
+            s3_misc.copy_object(self.akey, self.skey, self.src_bkt, "obj1", self.src_bkt, dest_obj)
+        except CTException as error:
+            self.log.info("Expected exception received %s", error)
+            assert_utils.assert_in(errmsg.S3_COPY_OBJECT_QUOTA_ERR, error.message, error)
+        self.log.info("Step 8:Perform Get API to get user and bucket stats and validate the object"
+                      "count and space utilization for user/bucket .")
+        self.log.debug("Perform & Verify GET API to get capacity usage stats")
+        self.verify_user_quota(self.akey, self.skey, self.user_id)
+        self.log.info("Step 9: Create one more user (user2)")
+        akey, skey, uid = self.create_iam_user()
+        self.log.info("Step 10: Create 2 buckets named 'src1' and 'dest1' under the user created"
+                      " in step 9.")
+        bucket_created = s3_misc.create_bucket(self.src_bkt, akey, skey)
+        assert bucket_created, "Failed to create bucket"
+        self.buckets_created.append([self.src_bkt, akey, skey])
+        bucket_created = s3_misc.create_bucket(self.dest_bkt1, akey, skey)
+        assert bucket_created, "Failed to create bucket"
+        self.buckets_created.append([self.dest_bkt1, akey, skey])
+        self.log.info("Step 11: Perform PUT API to set user level quota fields"
+                      " i.e. enabled(bool)=true, max_size(integer value)= -1 ,"
+                      " max_objects(integer value) = N .")
+        self.set_get_user_quota(config_dict, uid)
+        self.log.info("Step 12:Perform GET API to get user level quota fields and verify the"
+                      " user level quota fields as per above PUT request.")
+        self.verify_user_quota(akey, skey, uid)
+        self.log.info("Step 13: Upload a simple object obj1 of some random size S in src1")
+        random_size = self.csm_obj.random_gen.randrange(1, test_cfg["max_size"])
+        resp = s3_misc.create_put_objects("obj1", self.src_bkt, akey, skey,
+                                          object_size=random_size, block_size="1K")
+        assert resp, "Put object Failed"
+        self.log.info("Step 14: Perform Copy object operation within same bucket 'N-1' times using"
+                      "same source bucket and object and just changing the key name of the "
+                      "destination object .")
+        for obj_cnt in range(1, (test_cfg["max_objects"])):
```

Comment:
this range should be same as line 244?  
Can we do all these operations (for 2 users)using for loop, since all steps look alike

Reply:
In all tests, first scenario is mostly within same destination bucket while second ones are different bucket